### PR TITLE
Added validation_phase to results if there are findings

### LINKED ISSUE #142: Add validation_phase field to results
If validation errors are found during phase 1, add validation_phase with value "Syntactical" to all the findings.  If errors/warnings are found during phase 2, add validation_phase with value "Logical" to all the findings.

### PR DESCRIPTION
**Pull request by jcadam14**
Closes #142 

Had to pull in the index story #137 because adding the phase wasn't testing correctly.  So if this is approved before PR #138 then we can ignore that one.

### CODE REVIEW

File: src/regtech_data_validator/create_schemas.py [hunk 1]
```diff
@@ -177,6 +177,10 @@ def validate_phases(df: pd.DataFrame, context: dict[str, str] | None = None) ->
     p1_is_valid, p1_findings = validate(get_phase_1_schema_for_lei(context), df)
 
     if not p1_is_valid:
+        p1_findings.insert(1, "validation_phase", PHASE_1_TYPE, True)
         return p1_is_valid, p1_findings
```
Comment (lchen-2101) [inline src/regtech_data_validator/create_schemas.py:181]: I'm wondering if we could add the type in the return tuple as well; I was originally thinking of just adding it to the return tuple without modifying the findings object; and the downstream API can use the type to do with it what it wants.

Comment (lchen-2101) [reply src/regtech_data_validator/create_schemas.py:181]: oh right... insert does column insert... not row... nvm about not modifying the findings object

Comment (jcadam14) [reply src/regtech_data_validator/create_schemas.py:181]: We could do this instead.  I realized I need to at least set something either in the json, or through this approach, so the filing-api will know what to send in the submission object metadata.  Your call.  But yeah this is just adding a new column to the result set.  Only reason this approach may be desirable is if (and I know for MVP it's not a thing) we ever end up with a mixture of syntactical and logical errors going back, we'd want a designator on each finding, not just the return call.

Comment (lchen-2101) [reply src/regtech_data_validator/create_schemas.py:181]: yeap, I don't see the harm in having both, I can see utility in the downstream call to have one more thing in the return, and the column addition is good for the csv.

Comment (hkeeler) [reply src/regtech_data_validator/create_schemas.py:181]: Agreed. I like having the phase in the return tuple. Nice to be able to drive logic without having to look into the DataFrame.

Adding it to the DataFrame seems fine too. I don't think it hurts anything, unless we need to cut things later to optimize things for size or performance.

File: src/regtech_data_validator/phase_validations.py [hunk 2]
```diff
@@ -31,6 +31,9 @@
 )
 from regtech_data_validator.checks import SBLCheck, Severity
 
+PHASE_1_TYPE = "Syntactical"
+PHASE_2_TYPE = "Logical"
```
Comment (hkeeler) [inline src/regtech_data_validator/phase_validations.py:35]: These seem like a good candidate for an `enum`, similar to `Severity`. Maybe something like:

```python
class ValidationPhase(StrEnum):
    SYNTACTICAL = "Syntactical"
    LOGICAL = "Logical"
```

...and just abandon the whole "Phase 1/2" bit that doesn't have much meaning outside the internals of this codebase.

**Review by hkeeler — COMMENTED** [review]
Overall looks great. Just the one idea about switching phase stuff to an enum.

File: src/regtech_data_validator/phase_validations.py (hunk 2 above)
Comment (jcadam14) [reply src/regtech_data_validator/phase_validations.py]: Thought about making this an enum, shouldn't second guess myself.  Pushed changes.  Moved the enum to the file that uses it.

Comment (hkeeler) [reply src/regtech_data_validator/phase_validations.py]: 👍 Only odd numbered guesses allowed.

File: src/regtech_data_validator/create_schemas.py [hunk 3]
```diff
@@ -9,6 +9,13 @@
 from regtech_data_validator.phase_validations import get_phase_1_and_2_validations_for_lei
 from regtech_data_validator.schema_template import get_template
 
+from enum import Enum
+
+
+class ValidationPhase(str, Enum):
+    SYNTACTICAL = "Syntactical"
+    LOGICAL = "Logical"
```
Comment (hkeeler) [inline src/regtech_data_validator/create_schemas.py:17]: Elsewhere in the code we use [`enum.StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum). This _may_ end up being equivalent, but probably want to switch just for consistency.

https://github.com/cfpb/regtech-data-validator/blob/63b4735b85aca3837e7d7da0f587ac2580e0ab49/src/regtech_data_validator/checks.py#L13-L15

Comment (jcadam14) [reply src/regtech_data_validator/create_schemas.py:17]: Ah in all the other repos we do (str, Enum), StrEnum is a 3.11 and beyond thing so maybe that's why.  I'll swap here, might be good to move them all over to StrEnum at some point

**Review by hkeeler — APPROVED** [review]

**Pull request merged**